cidc_api/models/templates/sync_schemas.py

```diff
@@ -0,0 +1,153 @@
+from sqlalchemy.orm.session import Session
+from typing import Dict, List
+
+from ..models import TrialMetadata, UploadJobs
+from .trial_metadata import (
+    ClinicalTrial,
+    Cohort,
+    CollectionEvent,
+    Participant,
+    Sample,
+    Shipment,
+)
+from .utils import with_default_session
+
+
+def _make_sample_to_shipment_map(trial_id: str, session: Session) -> Dict[str, str]:
+    sample_to_shipment_map = {}
+
+    uploads = (
+        session.query(UploadJobs)
+        .filter(UploadJobs.trial_id == trial_id, UploadJobs.multifile == False)
+        .all()
+    )
+    for upload in uploads:
+        shipments = upload.metadata_patch.get("shipments", [])
+        assert (
+            len(shipments) == 1
+        ), f"Multiple/no shipments in single upload: {upload.id} on trial {upload.trial_id}"
+        manifest_id = shipments[0]["manifest_id"]
+
+        for partic in upload.metadata_patch["participants"]:
+            for sample in partic["samples"]:
+                sample_to_shipment_map[sample["cimac_id"]] = manifest_id
+
+    return sample_to_shipment_map
+
+
+@with_default_session
+def syncall_from_blobs(session: Session, dry_run: bool = False,) -> List[Exception]:
```

Comment:
@scvannost so this function seem wayyy long. It'll be hard to provide meaningful feedback on this without inline comments. Ideally this would be broken up into multiple functions with doc-strings describing the purpose.

Reply:
in next commit